vignettes/theme.Rmd

```diff
@@ -0,0 +1,321 @@
+
+### Custom Bootstrap themes
+
+You can also create your own custom theme by setting the [main colors and fonts](https://rstudio.github.io/bslib/articles/theming.html#main-colors-fonts) (e.g., `bg`, `fg`, `primary`, `base_font`, `code_font`, etc.) as well as more specific theming options, [such as `navbar-bg`](https://rstudio.github.io/bslib/articles/bs4-variables.html#navbar-bg). Under the hood, this works by passing the `theme` to `bslib::bs_theme()` which influences the final look of the page by creating a custom build of Bootstrap (and Flexdashboard) CSS.
+
+``` {.r}
+---
+output:
+  flexdashboard::flex_dashboard:
+    theme:
+      version: 4
+      bg: "#101010"
+      fg: "#FDF7F7" 
+      primary: "#ED79F9"
+      navbar-bg: "#3ADAC6"
+      base_font: 
+        google: Prompt
+      code_font:
+        google: JetBrains Mono
+---
+```
+
+![](images/theme-dark.png){.snippet}
+
+As shown above, a major benefit of theming via `{bslib}` is that theming of things like `gauge()`, `valueBox()`, and even `DT::datatable()` "just work" (that is, their styling defaults are informed by the `{bslib}` theme).
+
+These components work with `{bslib}`'s real-time theming widget, which you can add to `flex_dashboard` by adding either:
+
+-   `runtime: shiny` to the document's yaml matter and calling `bslib::bs_themer()` in any code chunk, or
+
+-   `runtime: shinyrmd` to the document's yaml matter and calling `bslib::bs_themer()` in a code chunk with `context='server'`:
+
+        ```{r setup, include = FALSE, context = "server"}`r ''`
+        bslib::bs_themer()
+        ```
+
+![](images/real-time.gif){.snippet}
+
+### Theme output
+
+Generally, most "un-opinionated" HTML content rendered by the web browser "just works" with `{bslib}`. This means that the styles you specify for your document will cascade correctly. However, some HTML elements that aren't rendered by the browser aren't necessarily aware of `{bslib}` (or CSS in general).
+
+To theme static R plots (e.g., `{ggplot2}`), you can call `thematic::thematic_rmd()` in a setup chunk to translate the `{bslib}` theme main colors to R plot theming defaults.
+
+Additionally, if you're using Google Fonts, or some other font that you've registered with R, you can safely include `font = "auto"` to translate fonts as well:
+
+    ```{r setup, include = FALSE}`r ''`
+    thematic::thematic_rmd(font = "auto")
+    ```
+
+![](images/theme-plot.png){.snippet}
+
+If you use one of the `{ggplot2}` [complete themes](https://ggplot2.tidyverse.org/reference/ggtheme.html) (like `theme_bw()` or `theme_minimal()`, for example), we recommend setting the theme globally for the `flex_dashboard` in a `setup` chunk, then turning thematic on *after*.
+
+    ```{r setup, include = FALSE}`r ''`
+    library(ggplot2)
+    theme_set(theme_minimal())
+    thematic::thematic_rmd(font = "auto")
+    ```
+    
+Read more about the `{thematic}` package here: 
+`vignette("custom", package = "thematic")`.
```

Comment:
Not sure why but there is no autolink to the thematic package. That is too bad as it could be possible. Maybe some requirements are missing for this: https://pkgdown.r-lib.org/articles/linking.html